### PR TITLE
feat: Make ServerVerticle handler configurable.

### PR DESCRIPTION
**Pull request by halber**
The `ServerVerticle` attaches request handlers to the route. The request handlers attached to the route cannot be changed at the moment. This change makes the request handlers that the `ServerVerticle` attaches to the route configurable.

A `RoutingHandlerFactory` is used to create an instance of the `Handler<RoutingContext>` class. In the `ServerConfig`, a list of `RoutingHandlerFactory` class names is used to specify the `RoutingHandlerFactory`. The handler created by the `RoutingHandlerFactory` is then added to the root route. The request handlers are added in the order specified in the `ServerConfig` list.

### CODE REVIEW

**Review by BenniR6 — COMMENTED** [review]
-No singletons needed for handlers as these are only used on start up.
-Setting the allow file upload to true is missing in this PR -> as already mentioned

**Documentation will follow.**

**Comment by halber** [comment]
> -Setting the allow file upload to true is missing in this PR -> as already mentioned
Maybe we have misunderstood each other. I will not add such a handler because at least in my option such ha handler is not required and can't be implemented easily because the implementation depends heavily on the thing needed.

**Comment by halber** [comment]
Do not merge! there is an issue I currently not understand.

**Comment by pk-work** [comment]
Concept was validated in the committers meeting - 2022-08-02

File: src/main/java/io/neonbee/internal/handler/factories/SessionHandlerFactory.java
Comment (s4heid) [inline src/main/java/io/neonbee/internal/handler/factories/SessionHandlerFactory.java:22]:
```suggestion
 * If ServerConfig.SessionHandling is set to NONE a no-operation handler is returned, which does not perform any session handling.
```

File: src/main/java/io/neonbee/internal/handler/factories/SessionHandlerFactory.java
Comment (s4heid) [inline src/main/java/io/neonbee/internal/handler/factories/SessionHandlerFactory.java:50]:
```suggestion
     * @return an optional session store, suitable for the given Vert.x instance and based on the provided config value
```

File: src/main/java/io/neonbee/internal/handler/factories/SessionHandlerFactory.java
Comment (s4heid) [inline src/main/java/io/neonbee/internal/handler/factories/SessionHandlerFactory.java:57]: Maybe we can move all these code comments into the javadoc where it belongs?

File: src/test/java/io/neonbee/internal/handler/factories/CorrelationIdHandlerFactoryTest.java
Comment (s4heid) [inline src/test/java/io/neonbee/internal/handler/factories/CorrelationIdHandlerFactoryTest.java:26]: I think you could also also work here with the NeonBeeMockHelper in order to avoid the usage of mockStatic. Not required to change this, but generally I think it's more convenient to use the helper.

**Review by s4heid — CHANGES_REQUESTED** [review]

File: docs/ServerVerticle.md
Comment (halber) [reply docs/ServerVerticle.md:11]: Unfortunately, I can't find any documentation from vertx for this.

**Review by s4heid — APPROVED** [review]

**Pull request merged**
